### PR TITLE
Add automatic RBAC creation for kubeletstats receiver

### PR DESCRIPTION
**Pull request by iblancasa**
**Link to tracking Issue(s):**
- Resolves: #3155

**Testing:** added unit and E2E tests.

### CODE REVIEW

File: internal/components/receivers/scraper_test.go [hunk 1]
```diff
@@ -29,7 +29,6 @@ func TestScraperParsers(t *testing.T) {
 		defaultPort  int
 	}{
 		{"prometheus", "__prometheus", 0},
-		{"kubeletstats", "__kubeletstats", 0},
```
Comment (jaronoff97) [inline internal/components/receivers/scraper_test.go:32]: this should still be a scraper right?

File: tests/e2e-automatic-rbac/receiver-kubeletstats/01-assert.yaml
Comment (jaronoff97) [inline tests/e2e-automatic-rbac/receiver-kubeletstats/01-assert.yaml:1]: should we just put the tests from this folder into the extra-permissions-operator test?

File: internal/components/receivers/kubeletstats.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		config.Metrics.K8sPodCPURequestUtilization.Enabled,
+		config.Metrics.K8sPodMemoryLimitUtilization.Enabled,
+		config.Metrics.K8sPodMemoryRequestUtilization.Enabled,
+	}
+	for _, metric := range metrics {
+		if metric {
+			prs = append(prs, nodesProxyPr)
+			return prs, nil
+		}
+	}
```
Comment (swiatekm) [inline internal/components/receivers/kubeletstats.go:83]: This is a bit less efficient, but imo more clear. WDYT?

```suggestion
        resourceMetricEnabled := false
	metricsEnabled := []bool{
		config.Metrics.K8sContainerCPULimitUtilization.Enabled,
		config.Metrics.K8sContainerCPURequestUtilization.Enabled,
		config.Metrics.K8sContainerMemoryLimitUtilization.Enabled,
		config.Metrics.K8sContainerMemoryRequestUtilization.Enabled,
		config.Metrics.K8sPodCPULimitUtilization.Enabled,
		config.Metrics.K8sPodCPURequestUtilization.Enabled,
		config.Metrics.K8sPodMemoryLimitUtilization.Enabled,
		config.Metrics.K8sPodMemoryRequestUtilization.Enabled,
	}
	for _, metricEnabled := range metricsEnabled {
		resourceMetricEnabled = resourceMetricEnabled || metricEnabled
	}
	
	if len(config.ExtraMetadataLabels) > 0 || resourceMetricEnabled {
	         prs = append(prs, nodesProxyPr)
	}
	
	return prs, nil
```

File: tests/e2e-automatic-rbac/receiver-kubeletstats/01-assert.yaml
Comment (iblancasa) [reply tests/e2e-automatic-rbac/receiver-kubeletstats/01-assert.yaml:1]: No. That folder contains some patchs for `kustomize` for the E2E tests.

File: internal/components/receivers/scraper_test.go (hunk 1 above)
Comment (iblancasa) [reply internal/components/receivers/scraper_test.go:32]: I don't think so. It was removed from being a scraper in `internal/components/receivers/helpers.go` and now is treated differently.

File: internal/components/receivers/kubeletstats.go (hunk 2 above)
Comment (iblancasa) [reply internal/components/receivers/kubeletstats.go:83]: Actually, it looks more confusing to me 😅. The current approach is like "if this thing is defined, just return". And now you need to read the full thing... but if you have a strong opinion about this, I can change it.

**Review by swiatekm — APPROVED** [review]

File: internal/components/receivers/scraper_test.go (hunk 1 above)
Comment (iblancasa) [reply internal/components/receivers/scraper_test.go]: Reverted

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**